### PR TITLE
Emcs 633 workitem completion

### PR DESCRIPTION
**Pull request by Froom2**
This PR adds the handling of the workitems depending on the response the connector gets from NRS.
Also adds logging in.

### CODE REVIEW

**Review by RobertLaverick — CHANGES_REQUESTED** [review]
Only one change really (if else if to match) the rest are questions

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/NrsConnectorNew.scala [hunk 1]
```diff
@@ -67,10 +69,23 @@ class NrsConnectorNew @Inject() (
...
-        } else {
-          Future.failed(UnexpectedResponseException(response.status, response.body))
+
+        response.status match {
+          case status if is5xx(status) =>
+            logger.warn(
+              s"Non repudiation submission failed with status ${response.status}"
+            )
+            Future.failed(UnexpectedResponseException(response.status, response.body))
+          case ACCEPTED =>
```
Comment (RobertLaverick) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/NrsConnectorNew.scala:79]: I'd possibly make this the first case (then it's good path then all bad paths)

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsServiceNew.scala [hunk 2]
```diff
@@ -76,11 +79,30 @@ class NrsServiceNew @Inject() (
...
-    } yield Done
+    nrsConnectorNew
+      .sendToNrs(workItem.item.payload, correlationIdService.generateCorrelationId())
+      .flatMap { _ =>
+        nrsWorkItemRepository
+          .complete(workItem.id, Succeeded)
+          .map(_ => Done)
+      }
+      .recoverWith {
+        case ex: UnexpectedResponseException if is4xx(ex.status) =>
```
Comment (RobertLaverick) [inline app/uk/gov/hmrc/excisemovementcontrolsystemapi/services/NrsServiceNew.scala:90]: I'm slightly concerned about this case being marked as permenantly failed. Might be me being over-cautious, but *in theory* any 4xx should be our fault, do we really want to never resubmit them? Then again multiple failures would also be lost so maybe this isn't an issue?

Maybe something to chat about post-review/pre-merge

File: app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/NrsConnectorNew.scala (hunk 1 above)
Comment (Froom2) [reply app/uk/gov/hmrc/excisemovementcontrolsystemapi/connectors/NrsConnectorNew.scala:79]: fair point!

**Pull request merged**